web/modules/custom/custom_multistep_form/src/Plugin/Block/FormConfigureBlock.php

```diff
@@ -0,0 +1,245 @@
+<?php
+
+namespace Drupal\custom_multistep_form\Plugin\Block;
+
+use Drupal\Core\Block\Attribute\Block;
+use Drupal\Core\Block\BlockBase;
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\Form\FormBuilderInterface;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\Core\StringTranslation\TranslatableMarkup;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Provides a 'FormConfigureBlock' block.
+ */
+#[Block(
+  id: "form_configure_block",
+  admin_label: new TranslatableMarkup("Configure Form Steps"),
+)]
+class FormConfigureBlock extends BlockBase implements ContainerFactoryPluginInterface {
+
+  /**
+   * Constructor for FormConfigureBlock.
+   */
+  public function __construct(array $configuration,
+    $plugin_id,
+    $plugin_definition,
+    protected ConfigFactoryInterface $configFactory,
+    protected FormBuilderInterface $formBuilder,
+  ) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('config.factory'),
+      $container->get('form_builder'),
+    );
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  public function build(): array {
+    return $this->formBuilder->getForm('\Drupal\custom_multistep_form\Form\CustomMultistepForm');
+  }
+
+  /**
+   * Configuration for Form.
+   */
+  public function blockForm($form, FormStateInterface $form_state): array {
+    $config = $this->configFactory->getEditable('custom_multistep_form.settings');
+
+    $form['steps'] = [
+      '#type' => 'details',
+      '#title' => $this->t('Form Steps Configuration'),
+      '#open' => TRUE,
+      '#prefix' => '<div id="custom-configuration-form">',
+      '#suffix' => '</div>',
+    ];
+
+    $form['steps']['step_order'] = [
+      '#type' => 'table',
+      '#header' => [$this->t('Step'), $this->t('Weight')],
+      '#empty' => $this->t('No steps available.'),
+    ];
+
+    $active_steps_count = 0;
+    for ($i = 1; $i <= 3; $i++) {
```

Comment:
it is VERY unclear again
If the config is empty, and it will be right after installing the config, where are the steps added?
And available steps anyway hardcoded in the form, so why?